src/exporters/s5.py

```diff
@@ -0,0 +1,492 @@
+                False
+            ), f'granularity must be in ["monthly", "hourly"]\
+                Currently: {self.granularity}'
+
+        # convert to strings
+        leadtime_times_str = [str(lt) for lt in leadtime_times]
+        selection_dict = {
+            leadtime_key: leadtime_times_str  # leadtime_key differs for monthly/hourly
+        }
+
+        return selection_dict
+
+    def get_product_type(self, product_type: Optional[str] = None) -> Optional[str]:
+        """By default download the 'monthly_mean' product for monthly data """
+        valid_product_types = self.dataset_reference["product_type"]
+
+        if valid_product_types is None:
+            f"{self.dataset} has no `product_type` key. \
+            Only the monthly datasets have product types!"
+            return None
+
+        # if not provided then download monthly_mean
+        if product_type is None:
+            print("No `product_type` provided. Therefore, downloading `monthly_mean`")
+            product_type = "monthly_mean"
+
+        assert (
+            product_type in valid_product_types
+        ), f"Invalid `product_type`: \
+            {product_type}. Must be one of: {valid_product_types}"
+
+        return product_type
+
+    # @staticmethod
+    # def _correct_input(value, key):
+    #     if type(value) is str:
+    #         # check the string is correctly formatted
+    #         if key == 'time':
+    #             assert (re.match(r"\d{2}:0{2}", value)), \
+    #                 f'Expected time string {value} to be in hour:minute format, \
+    #                 e.g. 01:00. Minutes MUST be `00`'
+    #         return value
+    #     else:
+    #         if key == 'year':
+    #             return str(value)
+    #         elif key in {'month', 'day'}:
+    #             return '{:02d}'.format(value)
+    #         elif key == 'time':
+    #             return '{:02d}:00'.format(value)
+    #     return str(value)
+
+    def get_pressure_levels(
+        self, pressure_levels: Optional[List[int]] = None
+    ) -> Optional[Dict]:
+        if self.pressure_level:
+            # make sure the dataset requires pressure_levels
+            assert "pressure_level" in [
+                k for k in self.dataset_reference.keys()
+            ], f"\
+            {self.dataset} has no 'pressure_level' keys. Cannot assign pressure levels"
+
+            # make sure the pressure levels are legitimate choices
+            assert all(
+                np.isin(pressure_levels, self.dataset_reference["pressure_level"])
+            ), f"\
+            {pressure_levels} contains invalid pressure levels!\
+            Must be one of:{self.dataset_reference['pressure_level']}"
+
+            if pressure_levels is not None:  # (s0rry) for mypy
+                return {"pressure_level": [str(pl) for pl in pressure_levels]}
+            else:
+                return None
+        else:
+            return None
+
+    def get_valid_variables(self) -> List:
```

Comment:
This can be a property:
```python
    @property
    def valid_variables(self) -> List:
        return self.dataset_reference["variable"]
```
This way, other functions can call it as
```python
valid_variables = self.valid_variables
```

Reply:
cool!